config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,113 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+type OpenStackAPIBGPPeer struct {
+	// asn is the ASN of the peer.
+	//
+	// +kubebuilder:validation:Required
+	ASN string `json:"asn"`
+
+	// ip is the IP address of the peer. It may be either IPv4 or IPv6.
```

Comment:
I think we can add validation for this but it needs to be done via a patch, the networking teams have some fields with openapi IP validations, we should investigate that.

Are there restrictions on when this could be v4 vs v6?

Reply:
I'm not aware of any restrictions. Even if we're IPv4-only internally, IPv6 should still work here because it's publishing an address to the upstream router. Assuming the address is correct and the upstream router is correctly configured, both of which are outside of our control, it should be just like any other traffic in.

Reply:
added `// +kubebuilder:validation:Format=ip`. Is that it?

Reply:
You have to use a patch like https://github.com/openshift/api/pull/1278/files#diff-8e2a77037de011306a487acc4a7bd2592d519f3415ad6c8b93d191d8cb3e3c32R1